src/indexer/merger.rs

```diff
@@ -624,18 +845,39 @@ impl IndexMerger {
                 while doc != TERMINATED {
                     // deleted doc are skipped as they do not have a `remapped_doc_id`.
                     if let Some(remapped_doc_id) = old_to_new_doc_id[doc as usize] {
-                        // we make sure to only write the term iff
+                        // we make sure to only write the term if
                         // there is at least one document.
                         let term_freq = segment_postings.term_freq();
                         segment_postings.positions(&mut positions_buffer);
-
-                        let delta_positions = delta_computer.compute_delta(&positions_buffer);
-                        field_serializer.write_doc(remapped_doc_id, term_freq, delta_positions);
+                        // if doc_id_mapping exists, the docids are reordered, they are
+                        // not just stacked. The field serializer expects monotonically increasing
+                        // docids, so we collect and sort them first, before writing.
+                        //
+                        // I think this is not strictly necessary, it would be possible to
```

Comment:
That sounds very expensive. Did you test it on a dataset?

Reply:
merging was 2x slower with sorting on a dataset I tested, but I have a change in the pipeline, which will increase merge speed in both cases :).
Currently the overhead of the sorting is the same as in `recorder.rs`

Reply:
0,9% for the merge time is spend in the vector push of the posting in the webserver dataset (tokenized url field with positions)